project_template/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX
+    VERSION 0.1.0)
+
+option(az_exampleshortname_with_docs "Build documentation" ON)

```

Comment:
Lowercase options feel "weird" to me (as someone who reads a lot of CMake files).

Reply:
I like them. Only downside is #cmakedefine